tests/Forms/EditFormFactoryTest.php

```diff
@@ -21,4 +24,42 @@ public function testFormFieldsHaveNamespaces()
 
         $this->assertNotNull($fields->dataFieldByName('PageElements_' . $record->ID . '_Title'));
     }
+
+    public function testNamespaceFields(): void
+    {
+        $record = $this->objFromFixture(ElementContent::class, 'content_block');
+        $factory = new EditFormFactory();
+        $fields = new FieldList([
+            new TextField('FieldOne'),
+            new LiteralField('IgnoredField', ''),
+            new UploadField('FieldTwo'),
+        ]);
+        $factory->namespaceFields($fields, ['Record' => $record]);
+
+        $expectedNames = [
+            'PageElements_' . $record->ID . '_FieldOne',
+            'IgnoredField',
+            'PageElements_' . $record->ID . '_FieldTwo',
+        ];
+        $this->assertSame($expectedNames, $fields->column('Name'));
+    }
+
+    public function testRemoveNamespaceFromFields(): void
+    {
+        $record = $this->objFromFixture(ElementContent::class, 'content_block');
+        $factory = new EditFormFactory();
+        $fields = new FieldList([
+            new TextField('PageElements_' . $record->ID . '_FieldOne'),
+            new LiteralField('IgnoredField', ''),
```

Comment:
This is how the literal field will look when passed in, so we're just including it here to validate it doesn't get mangled